### PR TITLE
Download SpringAI project as zip file

### PR DESCRIPTION
**Pull request by corradodebari**
SpringAI project download, supporting OpenAI/OLLAMA deployment. 
Include OpenAI API REST interface for chat completion. 
Not yet the full configuration available.

### CODE REVIEW

File: spring_ai/pom.xml [hunk 1]
```diff
@@ -0,0 +1,163 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+	<parent>
+		<groupId>org.springframework.boot</groupId>
+		<artifactId>spring-boot-starter-parent</artifactId>
+		<version>3.3.0</version>
```
Comment (andytael) [inline spring_ai/pom.xml:9]: This should be version 3.3.5

File: spring_ai/pom.xml [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+		<relativePath /> <!-- lookup parent from repository -->
+	</parent>
+	<groupId>com.example</groupId>
+	<artifactId>myspringai</artifactId>
+	<version>0.0.1-SNAPSHOT</version>
+	<name>myspringai</name>
+	<description>Simple AI Application using OpenAPI Service</description>
+	<properties>
+		<java.version>17</java.version>
```
Comment (andytael) [inline spring_ai/pom.xml:18]: Why not Java 21?

File: spring_ai/pom.xml [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+		</dependency>
+		<!--<dependency>
+			<groupId>org.springframework.ai</groupId>
+			<artifactId>spring-ai-oracle-store</artifactId>
+			<version>1.0.0-SNAPSHOT</version>
+		</dependency>-->
+		<dependency>
+			<groupId>com.oracle.database.jdbc</groupId>
+			<artifactId>ojdbc11</artifactId>
+			<version>23.5.0.24.07</version>
```
Comment (andytael) [inline spring_ai/pom.xml:87]: This shoul be using the spring boot starter:
        <dependency>
            <groupId>com.oracle.database.spring</groupId>
            <artifactId>oracle-spring-boot-starter-ucp</artifactId>
            <version>24.3.0</version>
        </dependency>

File: spring_ai/src/main/resources/application-dev.yml [hunk 4]
```diff
@@ -0,0 +1,50 @@
+server:
+  servlet:
+    context-path: /v1
```
Comment (andytael) [inline spring_ai/src/main/resources/application-dev.yml:3]: If you use Java 21, then add:

  threads:
    virtual:
      enabled: true

File: app/src/modules/st_common.py [hunk 5]
```diff
@@ -197,6 +208,223 @@ def initialize_chatbot(ll_model):
...
+
+    yaml_base=f"""
+server:
+  servlet:
+    context-path: /v1
+spring:
+  datasource:
+    url: ${{spring.datasource.url}}
+    username: ${{spring.datasource.username}}
+    password: ${{spring.datasource.password}}
```
Comment (andytael) [inline app/src/modules/st_common.py:235]: I think this code is the code that goes into `application.yaml` if so we should use a pooled connection:

```yaml
  datasource:
    url: ${spring.datasource.url}
    username: ${spring.datasource.username}
    password: ${spring.datasource.password}
    driver-class-name: oracle.jdbc.OracleDriver
    type: oracle.ucp.jdbc.PoolDataSource
    oracleucp:
      connection-factory-class-name: oracle.jdbc.pool.OracleDataSource
      connection-pool-name: UCPConnectionPool
      initial-pool-size: 15
      min-pool-size: 10
      max-pool-size: 30
```

File: spring_ai/pom.xml (hunk 2 above)
Comment (gotsysdba) [reply spring_ai/pom.xml:18]: For backward compat.

File: spring_ai/pom.xml (hunk 1 above)
Comment (gotsysdba) [reply spring_ai/pom.xml:9]: to be tested

File: app/src/modules/st_common.py (hunk 5 above)
Comment (gotsysdba) [reply app/src/modules/st_common.py:235]: checking

**Review by gotsysdba — APPROVED** [review]
LGTM

**Pull request merged**